vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -123,8 +144,20 @@ protected void setDisplayProperties(ILinkDisplay linkDisplay, String id, float p
             image.setParentPath(relativePath);
             imageRepo.save((VSImage) image);
             linkDisplay.setImage(image);
+        } 
+        
+    }
+    
+    protected void setDisplayProperties(ILinkDisplay linkDisplay, float positionX,float positionY,int rotation, DisplayType displayType, String existingImageId) throws ImageCouldNotBeStoredException, ImageDoesNotExistException {
+        linkDisplay.setPositionX(positionX);
+        linkDisplay.setPositionY(positionY);
+        linkDisplay.setRotation(rotation);
+        linkDisplay.setType(displayType != null ? displayType : DisplayType.ARROW);
```

Comment:
duplicate code; there should be an extra method that's called from both `setDisplayProperties` methods.